engine/src/test/java/io/camunda/zeebe/process/test/engine/EngineClientTest.java

```diff
@@ -68,10 +71,12 @@ class EngineClientTest {
 
   private ZeebeTestEngine zeebeEngine;
   private ZeebeClient zeebeClient;
+  private List<Intent> responseIntents;
 
   @BeforeEach
   void setupGrpcServer() {
-    zeebeEngine = EngineFactory.create();
+    responseIntents = new ArrayList<>();
+    zeebeEngine = EngineFactory.create(responseIntents::add);
```

Comment:
🔧 We'll create the listener for each test case here. While the impact is probably low, we only might want to activate that listener for the new test case.

Reply:
💭 I was looking at it, I am not sure how to do this in an easy and maintainable way tbh.

Reply:
We could create the engine without the listener in the general `@BeforeEach` setup like before and only recreate it in the specific test case with the listener, I guess?